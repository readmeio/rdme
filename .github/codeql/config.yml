paths-ignore:
  - '__tests__/**'
