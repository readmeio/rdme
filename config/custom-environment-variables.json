{ "host": "README_HOST", "hub": "README_HUB" }
