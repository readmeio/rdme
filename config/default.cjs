module.exports = {
  // eslint-disable-next-line global-require
  cli: require('../package.json').name,
  host: 'https://dash.readme.com',
  hub: 'https://{project}.readme.io', // this is only used for the `open` command
};
