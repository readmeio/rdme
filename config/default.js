module.exports = {
  // eslint-disable-next-line global-require
  cli: require('../package.json').name,
  host: 'https://dash.readme.com',
  hub: 'https://{project}.readme.io',
};
