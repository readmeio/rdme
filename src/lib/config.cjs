const path = require('path');

/**
 * We have to do this otherwise `require('config')` loads from the cwd where the user is running
 * `rdme` which wont be what we want.
 *
 * This file also has a CJS extension because the `config` library uses `require()` calls and
 * because `rdme` has `type: module` in its `package.json` Node wants **everything** to use
 * `import`.
 */
const configDir = process.env.NODE_CONFIG_DIR;
process.env.NODE_CONFIG_DIR = path.join(__dirname, '../../config');

const config = require('config');

process.env.NODE_CONFIG_DIR = configDir;

module.exports = config;
