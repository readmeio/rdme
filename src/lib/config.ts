import pkg from '../../package.json';

const config = {
  cli: pkg.name,
  host: 'https://dash.readme.com',
  hub: 'https://{project}.readme.io', // this is only used for the `open` command
} as const;

export default config;
