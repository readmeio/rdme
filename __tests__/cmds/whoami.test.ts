import { describe, afterEach, it, expect } from 'vitest';

import Command from '../../src/cmds/whoami';
import config from '../../src/lib/config';
import configStore from '../../src/lib/configstore';

const cmd = new Command();

describe('rdme whoami', () => {
  afterEach(() => {
    configStore.clear();
  });

  it('should error if user is not authenticated', () => {
    configStore.delete('email');
    configStore.delete('project');

    return expect(cmd.run({})).rejects.toStrictEqual(new Error(`Please login using \`${config.cli} login\`.`));
  });

  it('should return the authenticated user', () => {
    configStore.set('email', 'email@example.com');
    configStore.set('project', 'subdomain');

    return expect(cmd.run({})).resolves.toMatchSnapshot();
  });
});
