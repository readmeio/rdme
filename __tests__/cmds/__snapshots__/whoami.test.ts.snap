// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rdme whoami > should return the authenticated user 1`] = `"You are currently logged in as email@example.com to the subdomain project."`;
